client/starwhale/core/dataset/store.py

```diff
@@ -218,55 +226,88 @@ def __init__(
         # https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html
         self.extra_s3_configs = json.loads(os.environ.get("SW_S3_EXTRA_CONFIGS", "{}"))
 
+    def fits(self, bucket: str, endpoint_loc: str) -> bool:
+        return self.bucket == bucket and self.endpoint_loc == endpoint_loc
+
     def __str__(self) -> str:
         return f"endpoint[{self.endpoint}]-region[{self.region}]"
 
     __repr__ = __str__
 
     @classmethod
-    def from_uri(cls, uri: str, auth_name: str = "") -> S3Connection:
+    def from_uri(cls, uri: str) -> S3Connection:
         """make S3 Connection by uri
 
         uri:
             - s3://username:password@127.0.0.1:8000/bucket/key
             - s3://127.0.0.1:8000/bucket/key
         """
-        from .type import S3LinkAuth
-
-        uri = uri.strip()
-        if not uri or not uri.startswith(("s3://", "minio://")):
+        r = urlparse(uri.strip())
+        if not r.scheme or r.scheme not in cls.supported_schemes:
```

Comment:
`if not r.scheme or r.scheme not in cls.supported_schemes:`
-->
`if r.scheme not in cls.supported_schemes:`
?

if the URI does not contain schema, the urlparse function will return empty str for the schema attribute.

```
In [20]: _up = urlparse("11/a ")

In [21]: _up
Out[21]: ParseResult(scheme='', netloc='', path='11/a ', params='', query='', fragment='')
```

Reply:
> `if not r.scheme or r.scheme not in cls.supported_schemes:` --> `if r.scheme not in cls.supported_schemes:` ?
> 
> if the URI does not contain schema, the urlparse function will return empty str for the schema attribute.
> 
> ```
> In [20]: _up = urlparse("11/a ")
> 
> In [21]: _up
> Out[21]: ParseResult(scheme='', netloc='', path='11/a ', params='', query='', fragment='')
> ```

done